hal/armv7m/imxrt/10xx/imxrt10xx.c

```diff
@@ -2079,6 +2082,11 @@ unsigned int _imxrt_cpuid(void)
 
 void _imxrt_wdgReload(void)
 {
+#if defined(WATCHDOG) && defined(NDEBUG)
+	hal_cpuDisableInterrupts();
+	*(imxrt_common.rtwdog + rtwdog_cnt) = RTWDOG_REFRESH_KEY;
+	hal_cpuEnableInterrupts();
```

Comment:
We need to disable interrupts to do this? Why?

Reply:
According to i.MX RT1064 Processor Reference Manual:

Before starting the refresh sequence, disable the global
interrupts. Otherwise, an interrupt could effectively invalidate
the refresh sequence, if the interrupt occurs before the refresh
writes finish. After the sequence finishes, restore the global
interrupt control state.

(Section 57.3.2.2 Refreshing the Watchdog)